### PR TITLE
Fixes Randgalt/record-builder/#153

### PR DESCRIPTION
**Pull request by pawellabaj**
Fixes Randgalt/record-builder/#153

### CODE REVIEW

File: record-builder-processor/src/main/java/io/soabase/recordbuilder/processor/CollectionBuilderUtils.java [hunk 1]
```diff
@@ -298,7 +300,13 @@ private String disambiguateGeneratedMethodName(List<RecordClassType> recordCompo
 
     private MethodSpec buildShimMethod(String name, TypeName mainType, Class<?> abstractType,
             ParameterizedTypeName parameterizedType, TypeVariableName... typeVariables) {
-        var code = CodeBlock.of("return (o != null) ? $T.copyOf(o) : $T.of()", mainType, mainType);
+        CodeBlock code;
+        if (mainType.equals(setTypeName)) {
```
Comment (Randgalt) [inline record-builder-processor/src/main/java/io/soabase/recordbuilder/processor/CollectionBuilderUtils.java:304]: Are there backward compatibility concerns with this? We should probably put this behind an option. wdyt?

**Comment by pawellabaj** [comment]
@Randgalt option sounds OK

How it should look like?

**Comment by Randgalt** [comment]
You would add an option here: https://github.com/Randgalt/record-builder/blob/master/record-builder-core/src/main/java/io/soabase/recordbuilder/core/RecordBuilder.java#L68 - there are many examples in the code

**Comment by pawellabaj** [comment]
@Randgalt , what a timing? :-)

Please, see at my proposed option and latest changes in a code

**Comment by Randgalt** [comment]
I'll get to this this week. @pawellabaj can you merge the two commits please?

**Comment by pawellabaj** [comment]
> can you merge the two commits please?

@Randgalt , I squashed commits. I assumed you meant that, didn't you?

**Comment by Randgalt** [comment]
Looks good. Please run `mvn clean verify` locally. The auto-formatter will make a few changes. Commit and push those changes before we merge this.

**Comment by pawellabaj** [comment]
> Looks good. Please run `mvn clean verify` locally. The auto-formatter will make a few changes. Commit and push those changes before we merge this.

@Randgalt , done

**Comment by Randgalt** [comment]
Thanks for the PR

**Comment by pawellabaj** [comment]
Great, waiting for the new release

**Comment by Randgalt** [comment]
done

**Pull request merged**